internal/pkg/runtime/engines/singularity/container.go

```diff
@@ -1418,32 +1418,38 @@ func (c *container) addTmpMount(system *mount.System) error {
 				return nil
 			}
 
-			vartmpSource = "/var_tmp"
+			vartmpSource = "var_tmp"
 
 			workdir, err := filepath.Abs(filepath.Clean(workdir))
 			if err != nil {
 				sylog.Warningf("Can't determine absolute path of workdir %s", workdir)
 			}
 
-			tmpSource = workdir + tmpSource
-			vartmpSource = workdir + vartmpSource
+			tmpSource = filepath.Join(workdir, tmpSource)
+			vartmpSource = filepath.Join(workdir, vartmpSource)
 
-			if err := fs.MkdirAll(tmpSource, 0755); err != nil {
+			if err := fs.Mkdir(tmpSource, os.ModeSticky|0777); err != nil {
```

Comment:
Why the change from `Mkdirall` -> `Mkdir`?

Reply:
To avoid sticky bit on eventually create parent directory and additionally in the context if user specify a workdir we can safely assume the directory exists, so I changed for `Mkdir`